### ISSUE TITLE
Deprecation warning when eager loading blocks

### ISSUE THREAD

**Issue by mikewink** [spicywebau/craft-neo#525]
#### Description

I follow the doc about eager loading Neo blocks and it works, I think, but I end up with a deprecation warning every time `useMemoized` is doing its thing. Once deactivated the warning is gone. 

Deprecation warning:

```
The order element query param has been deprecated. Use orderBy instead.
```
```
/<path-to-project>/craft/vendor/yiisoft/yii2/base/ArrayableTrait.php:126
```

#### Steps to reproduce

That's the code I use to eager load the `entry.builder` (builder is my Neo field)

```twig
{% if entry is defined and entry is not null %}
    {% do craft.app.elements.eagerLoadElements(
        className(entry),
        [entry],
        ['builder']
    ) %}
{% endif %}
```

That the code for eager loading the blocks

```twig
{% if entry is defined and entry.builder is not null %}
    {% for block in entry.builder %}
        {% do block.useMemoized(entry.builder) %}
    {% endfor %}

    {% for block in entry.builder if block.level == 1 %}
        {{ builderBlock(block, entry, loop.index) }}
    {% endfor %}
{% endif %}
```

The warning will show up once I use `block.parent` for example or I want to use `block.children.all()`. It does not happen when I use `block.level`. I'm confused what I do wrong. :/

Thank you very much for any help or tip. 

#### Other information
- Neo version: 2.11.21
- Craft version: 3.7.24
- (if multi-site) What is the affected Neo field's propagation method: no multi-site
- Is eager-loading used? yes

**Comment by ttempleton** [comment]
It doesn't look like you're doing anything wrong with the eager loading. I'm assuming you have a field with the handle `order` (not necessarily being used by a Neo field), which can cause that deprecation warning in some cases across Craft (https://github.com/craftcms/cms/issues/4134) including when Neo blocks are being eager loaded. I'm not sure if we can do anything to avoid it, but I'll keep this issue open for now and have a look into it.

**Comment by mikewink** [comment]
I did a check search for any `order` string in my templates, but there are no strings related to ordering a field. I will strip down everything to the are bones and investigate some more too. Thank you for your insight and the Craft issue! I'll keep you posted.

And thank you very much for Neo in the first place! 🏆

**Comment by ttempleton** [comment]
Sorry for the delay — looks like we were able to avoid that deprecation warning when eager loading Neo blocks, and I've just released Neo 2.12.1 with that change.
